### PR TITLE
[FIX] website_forum: display escaped warning message

### PR DESCRIPTION
**Pull request by nboulif**
Steps to reproduce:

  - Install `website_forum` module
  - Go to Users, select Joel Willis user and set his karma to 0
  - Go to Website homepage
  - Login as Portal user (Joel Willis)
  - Click on Forum and select any post
  - Upvote the answer.

Issue:

  Warning message is displayed but html content tags/syntax
  is not escaped.

Solution:

  Use `Markup` from `web.utils` to escape html in message.

opw-2878582

### CODE REVIEW

File: addons/website_forum/static/src/js/website_forum.js [hunk 1]
```diff
@@ -255,10 +256,10 @@ publicWidget.registry.websiteForum = publicWidget.Widget.extend({
             if (forumID) {
                 notifOptions.messageIsHtml = true;
                 const linkLabel = _.escape(_t("Read the guidelines to know how to gain karma."));
-                notifOptions.message = `
+                notifOptions.message = Markup(`
                     ${_.escape(notifOptions.message)}<br/>
                     <a class="alert-link" href="/forum/${forumID}/faq">${linkLabel}</a>
-                `;
+                `);
```
Comment (nle-odoo) [inline addons/website_forum/static/src/js/website_forum.js:262]:
```suggestion
                const linkLabel = _t("Read the guidelines to know how to gain karma.");
                notifOptions.message = Markup`
                    ${notifOptions.message}<br/>
                    <a class="alert-link" href="/forum/${forumID}/faq">${linkLabel}</a>
                `;
```
from [the Markup code](https://github.com/odoo/odoo/blob/c169c9d7db098e2c09fe08f76fcbe7d5cb7b82b4/addons/web/static/src/legacy/js/core/utils.js#L270-L272), you can drop the parenthesis and then what is not a markup variable (3 variables here I assume) will be escaped. This should have exactly the same effect as your current code (only `forumID` will be escaped too but it should be done anyway).

**Review by nle-odoo — APPROVED** [review]
looks good to me

File: addons/website_forum/static/src/js/website_forum.js (hunk 1 above)
Comment (nboulif) [reply addons/website_forum/static/src/js/website_forum.js]: Updated :+1: Thanks!

**Comment by rdeodoo** [comment]
(note that this is fixed somewhere between 15.0 and master, don't forgot to investigate and probably add a `up to` here?)

**Comment by nboulif** [comment]
Hello @rdeodoo, it was indeed changed in saas-15.3 (https://github.com/odoo/odoo/commit/56cc3dfab156f21c7ef97b3c407f1480b094fe93), thanks!
@fw-bot up to saas-15.2

**Comment by nle-odoo** [comment]
For odoo-rd/security: the justification of using Markup is that we are escaping all variable inside the template, and this is now exactly the same code that is already in saas-15.3 from https://github.com/odoo/odoo/commit/56cc3dfab156f21c7ef97b3c407f1480b094fe93

**Comment by nboulif** [comment]
@mart-e Thank you!
@robodoo r+

**Pull request closed without merging**